src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,170 @@
+            if (methodContext is not null)
+            {
+                newMember = newMember.FormatForExtraction(methodContext.Syntax);
+            }
+
+            // Insert the new extracted member before the first member
+            SyntaxNode firstMemberNode = await members.First().DeclaringSyntaxReferences.First()
+                .GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            editor.InsertBefore(generator.GetDeclaration(firstMemberNode), newMember);
+
+            // Replace argument with a reference to our new member
+            SyntaxNode identifier = generator.IdentifierName(newMemberName);
+            if (isInvoked)
+            {
+                editor.ReplaceNode(node, generator.WithExpression(identifier, node));
+            }
+            else
+            {
+                editor.ReplaceNode(node, generator.Argument(identifier)
+                    .WithTrailingTrivia(arrayArgument.Syntax.GetTrailingTrivia())); // add any extra trivia that was after the original argument
+            }
+
+            // Return changed document
+            return editor.GetChangedDocument();
+        }
+
+        private static IArrayCreationOperation GetArrayCreationOperation(SyntaxNode node, SemanticModel model, CancellationToken cancellationToken,
+                out bool isInvoked)
+        {
+            // The analyzer only passes a diagnostic for two scenarios, each having an IArrayCreationOperation:
+            //      1. The node is an IArgumentOperation that is a direct parent of an IArrayCreationOperation
+            //      2. The node is an IArrayCreationOperation already, as it was pulled from an
+            //         invocation, like with LINQ extension methods
+
+            // If this is a LINQ invocation, the node is already an IArrayCreationOperation
+            if (model.GetOperation(node, cancellationToken) is IArrayCreationOperation arrayCreation)
+            {
+                isInvoked = true;
+                return arrayCreation;
+            }
+            // Otherwise, we'll get the IArrayCreationOperation from the argument node's child
+            isInvoked = false;
+            return (IArrayCreationOperation)model.GetOperation(node.ChildNodes().First(), cancellationToken);
+        }
+
+        private static string GetExtractedMemberName(IEnumerable<string> memberNames, string parameterName)
+        {
+            bool hasCollectionEnding = s_collectionMemberEndings.Any(x => parameterName.EndsWith(x, true, CultureInfo.InvariantCulture));
+
+            if (parameterName == "source" // for LINQ, "sourceArray" is clearer than "source"
+                || (memberNames.Contains(parameterName) && !hasCollectionEnding))
+            {
+                parameterName += "Array";
+            }
+
+            if (memberNames.Contains(parameterName))
+            {
+                int suffix = 0;
+                while (memberNames.Contains(parameterName + suffix))
```

Comment:
Probably unlikely that it fails but maybe we should also put a condition on suffix.

Reply:
What do you mean?

Reply:
Sorry I meant that there is no upper bound check for the int value of suffix. As I said, unlikely we would exceed int value.

Reply:
Should I make `suffix` a `uint` type?

Reply:
No I think that's ok, if we exceed int limit that means we are in a really weird scenario so it's probably ok to fail.